src/asmcnc/apps/wifi_app/screen_wifi.py

```diff
@@ -503,6 +525,7 @@ def connect_wifi(self):
                 os.system('echo "update_config=1" | sudo tee --append /etc/wpa_supplicant/wpa_supplicant-wlan0.conf')
                 os.system('echo "country="' + self.country.text + '| sudo tee --append /etc/wpa_supplicant/wpa_supplicant-wlan0.conf')
 
+                os.system('sudo sed -i "s/wifi_connected_before=False/wifi_connected_before=True/" config.txt')

```

Comment:
Rather than repeating this line for each case, is it worth just having it once at the same place the network gets restarted (or after, if it's successful)?